tests/FSharp.Compiler.ComponentTests/Language/AttributeCheckingTests.fs

```diff
@@ -533,4 +533,43 @@ b.text("Hello 2") |> ignore
         |> shouldFail
         |> withDiagnostics [
             (Error 101, Line 16, Col 1, Line 16, Col 7, "This construct is deprecated. Use B instead")
-        ]
\ No newline at end of file
+        ]
+
+#if !NETCOREAPP
+    [<Fact(Skip = "NET472 is unsupported runtime for this kind of test.")>]
+#else
+    [<Fact>]
+#endif
+    let ``Regression: typechecker does not fail when attribute is on type variable (https://github.com/dotnet/fsharp/issues/13525)`` () =
+        let csharpBaseClass = 
+            CSharp """
+        using System.Diagnostics.CodeAnalysis;
+        
+        namespace CSharp
+        {
+        
+            public interface ITreeNode
+            {
+            }
+        
+            public static class Extensions
+            {
+                public static TNode Copy<TNode>([NotNull] this TNode node, ITreeNode context1 = null) where TNode : ITreeNode =>
```

Comment:
I see, so the repro has to have both an attribute and a variable type. Makes sense, and also explains why we haven't encountered this before.

Reply:
Yep